programs/clearing_house/src/math/margin.rs

```diff
@@ -185,12 +179,66 @@ pub fn calculate_perp_position_value_and_pnl(
     .checked_div(AMM_TO_QUOTE_PRECISION_RATIO_I128)
     .ok_or_else(math_error!())?;
 
-    let total_unsettled_pnl = unrealized_pnl
-        .checked_add(unrealized_funding)
-        .ok_or_else(math_error!())?;
+    let market_position = if market_position.is_lp() {
+        // compute lp metrics
+        let lp_metrics = compute_settle_lp_metrics(&market.amm, market_position)?;
 
-    let unsettled_asset_weight =
-        market.get_unsettled_asset_weight(total_unsettled_pnl, margin_requirement_type)?;
+        // compute standardized + dust position in baa/qaa
+        let dust_unsettled_pnl = -lp_metrics
```

Comment:
think including the quote remainder change in other comment, what we want here is the margin value of the too small base asset amount to be deducted (down only) like you did here

Reply:
```
// compute standardized + dust position in baa/qaa
        let dust_base_asset_value = calculate_base_asset_value_with_oracle_price(
            lp_metrics.remainder_base_asset_amount,
            oracle_price_data.price,
        )?
        .checked_add(1)
        .ok_or_else(math_error!())?;
```